data_model/src/events/data/filters.rs

```diff
@@ -1,192 +1,734 @@
-//! This module contains `EventFilter` and entities for filter
+//! This module contains filters for data events.
+//!
+//! For each event in [`super::events`], there's a corresponding filter type in this module. It can filter the events by origin id and event type.
+//!
+//! Event types are filtered with an `EventSet` type, allowing to filter for multiple event types at once.
 
 use core::fmt::Debug;
 
-use derive_more::Constructor;
+use getset::Getters;
 use iroha_data_model_derive::model;
 
 pub use self::model::*;
 use super::*;
 
-/// Filter for all events
-pub type DataEventFilter = FilterOpt<DataEntityFilter>;
-
 #[model]
 pub mod model {
     use super::*;
 
-    /// Optional filter. May pass all items or may filter them by `F`
-    ///
-    /// It's better than `Optional<F>` because `Optional` already has its own `filter` method and it
-    /// would be ugly to use fully qualified syntax to call `Filter::filter()` method on it.
-    /// Also `FilterOpt` variant names look better for filter needs
     #[derive(
         Debug,
         Clone,
         PartialEq,
         Eq,
         PartialOrd,
         Ord,
+        FromVariant,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+    )]
+    pub enum DataEventFilter {
```

Comment:
it is not possible to match multiple different `DataEvent`s?

Reply:
like if I wanted to match both all Account and Domain events at the same time

Reply:
Yeah, I see utility here. I guess I can make into a struct instead of enum, so each filter would be applied independently. Need to think about the API though

Reply:
This would be quite big in scope, so opened a separate issue for it: https://github.com/hyperledger/iroha/issues/4356